### PR TITLE
Add the ProducerSideService module

### PR DESCRIPTION
**Pull request by vickyliu-go4it**
With this change, we moved `ProducerSideServiceImpl` and
`ProducerSideServer` into the `ProducerSideService` module such that the
cloud collector will be able to reuse this code.

Bug: http://b/216085824

### CODE REVIEW

File: src/Service/CMakeLists.txt [hunk 1]
```diff
@@ -13,18 +13,15 @@ target_sources(OrbitServiceLib PRIVATE
...
-        ProducerSideServiceImpl.h)
+        OrbitService.h)
 
 target_include_directories(OrbitServiceLib PUBLIC ${CMAKE_CURRENT_SOURCE_DIR})
 
 target_link_libraries(OrbitServiceLib PUBLIC
         GrpcProtos
         OrbitVersion
         ProducerSideChannel
+        ProducerSideService
```
Comment (beckerhe) [inline src/Service/CMakeLists.txt:24]: Can you also create a `BUILD` file and add it as a dependency in `src/Service/BUILD` such that the internal build works?

File: src/ProducerSideService/include/ProducerSideService/ProducerSideServer.h [hunk 2]
```diff
@@ -1,9 +1,9 @@
-// Copyright (c) 2020 The Orbit Authors. All rights reserved.
+// Copyright (c) 2022 The Orbit Authors. All rights reserved.
 // Use of this source code is governed by a BSD-style license that can be
 // found in the LICENSE file.
 
-#ifndef ORBIT_SERVICE_TARGET_SIDE_SERVER_H_
-#define ORBIT_SERVICE_TARGET_SIDE_SERVER_H_
+#ifndef ORBIT_PRODUCER_SIDE_SERVICE_TARGET_SIDE_SERVER_H_
+#define ORBIT_PRODUCER_SIDE_SERVICE_TARGET_SIDE_SERVER_H_
```
Comment (dpallotti) [inline src/ProducerSideService/include/ProducerSideService/ProducerSideServer.h:6]: Ehm, I guess this shows how I had originally called the class... Could you adjust to `ORBIT_PRODUCER_SIDE_SERVICE_PRODUCER_SIDE_SERVER_H_` (also the comment at the end of the file)?

**Review by dpallotti — APPROVED** [review]
Well, what Henning said, the rest looks good.

File: src/Service/CMakeLists.txt (hunk 1 above)
Comment (vickyliu-go4it) [reply src/Service/CMakeLists.txt:24]: Done! Thanks for pointing this out.

File: src/ProducerSideService/include/ProducerSideService/ProducerSideServer.h (hunk 2 above)
Comment (vickyliu-go4it) [reply src/ProducerSideService/include/ProducerSideService/ProducerSideServer.h]: Fixed accordingly! Thanks!

**Review by beckerhe — COMMENTED** [review]
At first glance LGTM but can you do the safety review for the copybara import first to see if it works?

File: src/ProducerSideService/BUILD [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+        "@com_github_grpc_grpc//:grpc",
+        "@com_github_grpc_grpc//:grpc++",
+        "@com_google_absl//absl/base",
+        "@com_google_absl//absl/base:core_headers",
+        "@com_google_absl//absl/container:flat_hash_set",
+        "@com_google_absl//absl/flags:flag",
+        "@com_google_absl//absl/strings",
+        "@com_google_absl//absl/strings:str_format",
+        "@com_google_absl//absl/synchronization",
+        "@com_google_absl//absl/time",
```
Comment (beckerhe) [inline src/ProducerSideService/BUILD:30]: It seems this also needs to depend on `"@com_google_protobuf//:arena",`

Comment (beckerhe) [reply src/ProducerSideService/BUILD:30]: Nice. It works. Thanks!

**Comment by vickyliu-go4it** [comment]
> At first glance LGTM but can you do the safety review for the copybara import first to see if it works?

Done! Thanks for the suggestion and help!

**Pull request merged**